mypy/errorcodes.py

```diff
@@ -192,6 +192,12 @@ def __str__(self) -> str:
     "General",
     default_enabled=False,
 )
+USE_BEFORE_DEF: Final[ErrorCode] = ErrorCode(
+    "use-before-def",
+    "Warn about variables that are used before they are defined",
+    "General",
+    default_enabled=False,
```

Comment:
We should make this one _enabled_ by default.

Reply:
I think that we need to understand the performance of the definedness analysis before enabling it by default. Also the implementation is still somewhat incomplete, and it would be good to fix known issues first.

In particular, the current implementation of try statements (#14114, currently not merged) seems to be exponential which might become expensive. Perhaps there is a more efficient way to do the same thing.

Reply:
We all agreed that always-undefined should be enabled by default in 1.0. Then there is no point in postponing this now, the sooner we enable this, the sooner we get some feedback, and will be able to fix possible problems. We will at least get feedback from `mypy_primer`.

Reply:
A follow-up PR can be immediately created after this has been merged that turns this on by default to get feedback from mypy_primer. We don't need to wait until everything is complete (but we may want to wait for some additional changes before merging the PR).

Reply:
OK, let's do it this way then.